web/webserver.go

```diff
@@ -50,3 +52,17 @@ func (s *QuickFeedService) RegisterRouter(tm *auth.TokenManager, authConfig *oau
 
 	return router
 }
+
+// controller is a wrapper for the QuickFeedService handler that sets a write deadline for the submission stream.
+// TODO: Remove this when connect-go finally supports deadlines.
+// TODO: https://github.com/connectrpc/connect-go/issues/604
+func controller(h http.Handler) http.Handler {
+	timeout := 15 * time.Minute
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == qfconnect.QuickFeedServiceSubmissionStreamProcedure {
+			control := http.NewResponseController(w)
+			control.SetWriteDeadline(time.Now().Add(timeout))
```

Comment:
What should we do if this returns an error? If there is nothing to do, then we should do `_ = control.SetWriteDeadline...`